File: Placement Sheet/Arrays/Service Based/ReverseTheArray.java
import java.util.Scanner;

public class ReverseTheArray {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the size of the array: ");
        int size = input.nextInt();
        int[] arr = new int[size];

        for(int i = 0; i<size; i++) {
            System.out.print("Enter the element of the array: ");
            arr[i] = input.nextInt();
        }

        System.out.print("Enter the position: ");
        int ps = input.nextInt();
        input.close();
        System.out.println("..........Original Array.........");
        printArray(arr);
        reverseTheArray(arr, ps);
        System.out.println("..........Reversed Array..........");
        printArray(arr);
    }

    private static void printArray(int[] arr) {
        for (int i: arr) {
            System.out.print(i+" ");
        }
    }

    private static void reverseTheArray(int[]  arr, int m) {
        if (arr == null || arr.length == 0 || arr.length <= m) {
            return;
        }
        int start = m;
        int end = arr.length - 1;
        while (start <= end) {
            int temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;
            start++;
            end--;
        }
    }
}

File: Placement Sheet/Arrays/Service Based/ReverseTheArray.py
def reverseTheArray(arr, m):
    if arr is None or len(arr) == 0 or len(arr) <= m:
        return

    start = m + 1
    end = len(arr) - 1

    while start <= end:
        temp = arr[start]
        arr[start] = arr[end]
        arr[end] = temp
        start += 1
        end -= 1

size = int(input("Enter the size of the array: "))
arr = []

for i in range(size):
    data = int(input("Enter the data: "))
    arr.append(data)

ps = int(input("Enter the position: "))

print("..........Original Array..........")
print(arr)

print("..........Reversed Array..........")
reverseTheArray(arr, ps)
print(arr)